### PR TITLE
use explicit optionspec class

### PR DESCRIPTION
**Pull request by valentinsulzer**
Fixes issue in pipeline with passing a list that is supposed to be a list

### CODE REVIEW

File: iwutil/__init__.py [hunk 1]
```diff
@@ -30,6 +30,27 @@ def subplots_autolayout(
...
+        can be anything. If a list, the value must be in this list.
+    """
+    def __init__(self, default_value, other_allowed_values=None):
+        self.default_value = default_value
+        if other_allowed_values is not None:
+            self.allowed_values = [self.default_value]+other_allowed_values
+            self.any_value_allowed = False
+        else:
+            self.allowed_values = None
+            self.any_value_allowed = True
```
Comment (kratman) [inline iwutil/__init__.py:52]:
```suggestion
    def __init__(self, default_value, other_allowed_values: list | None =None):
        self.default_value = default_value
        self.allowed_values = None
        self.any_value_allowed = True
        if other_allowed_values is not None:
            self.allowed_values = [self.default_value]+other_allowed_values
            self.any_value_allowed = False
```

Comment (kratman) [reply iwutil/__init__.py:52]: This forces `other_allowed_values` to be a list and simplifies the code a bit

**Review by kratman — CHANGES_REQUESTED** [review]

**Review by kratman — APPROVED** [review]

**Comment by valentinsulzer** [comment]
@kratman working late!

**Comment by kratman** [comment]
> @kratman working late!

Heard my phone beep and I knew it was an easy review to finish

**Pull request merged**